repo/manifest/committed_manifest_manager.go

```diff
@@ -216,7 +223,9 @@ func (m *committedManifestManager) compact(ctx context.Context) error {
 func (m *committedManifestManager) maybeCompactLocked(ctx context.Context) error {
 	m.verifyLocked()
 
-	if len(m.committedContentIDs) < autoCompactionContentCount {
+	// Don't attempt to compact manifests if the repo was opened in read only mode
+	// since we'll just end up failing.
+	if m.readOnly || len(m.committedContentIDs) < autoCompactionContentCount {
```

Comment:
I think we should determine readonly-ness from the underlying content manager, which should in turn read it from the underlying blob.Storage.

Reply:
seems reasonable. I've opened #3230 to expose that information. If that looks fine I can switch this and #3225 to use that instead of passing a value from the local client config down the stack